sympy/integrals/transforms.py

```diff
@@ -1181,8 +1182,13 @@ def laplace_transform(f, t, s, **hints):
     inverse_laplace_transform, mellin_transform, fourier_transform
     hankel_transform, inverse_hankel_transform
     """
+
     if isinstance(f, MatrixBase) and hasattr(f, 'applyfunc'):
-        return f.applyfunc(lambda fij: laplace_transform(fij, t, s, **hints))
+        elements_trans = [laplace_transform(fij, t, s, **hints) for fij in f]
+        elements, avals, conditions = zip(*elements_trans)
+        f_laplace = type(f)(*f.shape, elements)
+        return f_laplace, Max(*avals), And(*conditions)
```

Comment:
This can't be done until the deprecation is complete, can it?

Reply:
This is just fixing something that is completely broken. On master we have:
```python
In [1]: M = Matrix([[t, 0], [0, t]])

In [2]: t, s = symbols('t, s')

In [3]: M = Matrix([[t, 0], [0, t]])

In [4]: laplace_transform(M, t, s)
Out[4]: 
⎡⎛1          ⎞               ⎤
⎢⎜──, 0, True⎟  (0, -∞, True)⎥
⎢⎜ 2         ⎟               ⎥
⎢⎝s          ⎠               ⎥
⎢                            ⎥
⎢               ⎛1          ⎞⎥
⎢(0, -∞, True)  ⎜──, 0, True⎟⎥
⎢               ⎜ 2         ⎟⎥
⎣               ⎝s          ⎠⎦
```
That's a Matrix but with each element being a tuple.

The PR gives:
```python
In [3]: laplace_transform(M, t, s)
Out[3]: 
⎛⎡1     ⎤         ⎞
⎜⎢──  0 ⎥         ⎟
⎜⎢ 2    ⎥         ⎟
⎜⎢s     ⎥         ⎟
⎜⎢      ⎥, 0, True⎟
⎜⎢    1 ⎥         ⎟
⎜⎢0   ──⎥         ⎟
⎜⎢     2⎥         ⎟
⎝⎣    s ⎦         ⎠
```
IOW the PR returns the matrix plus the conditions. On master you get a matrix of tuples with the conditions as part of the tuples in the elements of the matrix. That's so broken that I don't think any deprecation is needed.

Reply:
It's not broken for the person that knows this and looks at a given matrix element's tuple to get the condition(s). Not that I don't want to see the change -- I think it makes much more sense like this -- but this is backwards incompatible for anyone that knows and handles this already. So should the cycle be 1) we need a keyword that allows the new output but default to the old. Then when the deprecation occurs, 2) the keyword has to default to new style and be put into deprecation, and finally 3) the keyword gets dropped.

Reply:
I'm not sure this needs a deprecation cycle but I've added it in.

I realised when looking at it that actually on master reasonable results can be obtained when using `noconds=False` but this PR broke that behaviour. I've fixed that and added tests for it.